docs/data/material/components/grid2/grid2.md

```diff
@@ -0,0 +1,245 @@
+
+{{"demo": "AutoGrid.js", "bg": true}}
+
+### Variable width content
+
+Set one of the size breakpoint props to `"auto"` instead of `true` / a `number` to size
+a column based on the natural width of its content.
+
+{{"demo": "VariableWidthGrid.js", "bg": true}}
+
+## Nested Grid
+
+The grid container that renders inside another grid container is a nested grid which inherits the [`columns`](#columns) and [`spacing`](#spacing) from the top. The deep nested grid will inherit the props from the upper nested grid if it receives those props.
+
+{{"demo": "NestedGrid.js", "bg": true}}
+
+## Columns
+
+You can change the default number of columns (12) with the `columns` prop.
+
+{{"demo": "ColumnsGrid.js", "bg": true}}
+
+## Offset
+
+Move the item to the right by using offset props which can be:
+
+- number, for example, `mdOffset={2}` - when used the item is moved to the right by 2 columns starts from `md` breakpoint and up.
+- `"auto"` - when used, the item is moved to the right edge of the grid container.
+
+{{"demo": "OffsetGrid.js", "bg": true}}
+
+## Custom breakpoints
+
+If you specify custom breakpoints to the theme, you can use those names as grid item props in responsive values.
+
+```js
+import { ThemeProvider, createTheme } from '@mui/material/styles';
+
+function Demo() {
+  return (
+    <ThemeProvider
+      theme={createTheme({
+        breakpoints: {
+          values: {
+            laptop: 1024,
+            tablet: 640,
+            mobile: 0,
+            desktop: 1280,
+          },
+        },
+      })}
+    >
+      <Grid container spacing={{ mobile: 1, tablet: 2, laptop: 3 }}>
+        {Array.from(Array(4)).map((_, index) => (
+          <Grid mobile={6} tablet={4} laptop={3} key={index}>
+            <div>{index + 1}</div>
+          </Grid>
+        ))}
+      </Grid>
+    </ThemeProvider>
+  );
+}
+```
+
+:::info
+The custom breakpoints affect both the size and offset props:
+
+```diff
+- <Grid xs={6} xsOffset={2}>
++ <Grid mobile={6} mobileOffset={2}>
+```
+
+:::
+
+### TypeScript
+
+You have to set module augmentation on the theme breakpoints interface. The properties with `true` value will appear as `{key}`(size prop) and `{key}Offset`(offset prop).
+
+```ts
+declare module '@mui/system' {
+  interface BreakpointOverrides {
+    // Your custom breakpoints
+    laptop: true;
+    tablet: true;
+    mobile: true;
+    desktop: true;
+    // Remove default breakpoints
+    xs: false;
+    sm: false;
+    md: false;
+    lg: false;
+    xl: false;
+  }
+}
+```
+
+## Prevent scrollbar
+
+If you use grid as a container in a small viewport, you might see a horizontal scrollbar because the negative margin is applied on all sides of the grid container.
+
+To prevent the scrollbar, set `disableEqualOverflow` prop to `true`. It will enable negative margin only on the top and left sides of the grid which remove overflow on the right-hand side.
```

Comment:
```suggestion
To disable this scrollbar, set the `disableEqualOverflow` prop to `true`. 
This removes the negative margins from the bottom and right sides of the grid to prevent overflow.

The demo below shows how this works:
```